File: solutionFiles/ContainerWithMostWater.py
import json
import sys

def max_area(height):
    left, right = 0, len(height) - 1
    max_area = 0

    while left < right:
        width = right - left
        h = min(height[left], height[right])
        current_area = width * h
        max_area = max(max_area, current_area)

        if height[left] < height[right]:
            left += 1
        else:
            right -= 1

    return max_area

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python script.py '<json_string>'")
        sys.exit(1)

    # Parse the JSON input
    input_json = sys.argv[1]
    try:
        data = json.loads(input_json)
        height = data["height"]
    except (json.JSONDecodeError, KeyError):
        print("Invalid JSON input")
        sys.exit(1)

    # Compute the maximum area
    result = max_area(height)

    # Convert result to JSON string with the specified format
    output_json = json.dumps({"result": result})

    # Print the JSON string
    print(output_json)

File: solutionFiles/ContainerWithMostWater.php
<?php
function maxArea($height) {
    $left = 0;
    $right = count($height) - 1;
    $max_area = 0;

    while ($left < $right) {
        $width = $right - $left;
        $h = min($height[$left], $height[$right]);
        $current_area = $width * $h;
        $max_area = max($max_area, $current_area);

        if ($height[$left] < $height[$right]) {
            $left++;
        } else {
            $right--;
        }
    }

    return $max_area;
}

if ($argc != 2) {
    echo "Usage: php script.php '<json_string>'\n";
    exit(1);
}

// Parse the JSON input
$input_json = $argv[1];
$data = json_decode($input_json, true);

if (json_last_error() !== JSON_ERROR_NONE || !isset($data['height'])) {
    echo "Invalid JSON input\n";
    exit(1);
}

$height = $data['height'];

if (!is_array($height)) {
    echo "Invalid height data\n";
    exit(1);
}

// Compute the maximum area
$result = maxArea($height);

// Convert result to JSON string with the specified format
$output_json = json_encode(["result" => $result]);

// Print the JSON string
echo $output_json . "\n";
?>
